### PR TITLE
Sd 27 tests

### PR DESCRIPTION
**Pull request by BrainIsDead**
Resolve SD-27:
- Added `link_preview` import in `models.py`
- Modified `get_preview` method in `Link` model to return `LinkPreview` object
- Added `SignalListViewTest` class in `tests.py`
- Added test methods for `SignalListView` in `tests.py`
- Modified `get_queryset` method in `SignalsListView` to call `get_base_diffs` for each object in the queryset
- Added `get_context_data` method in `SignalsListView`
- Modified `name` and `display_name` fields in `DataSourceFactory` to use a sequence with a prefix.
- Modified `name` and `display_name` fields in `SourceSubdivisionFactory` to use a sequence with a prefix.
- Modified `description` field in `SourceSubdivisionFactory` to generate fake text with a maximum of 250 characters.
- Modified `name` field in `GeographyFactory` to use a sequence with a prefix.
- Modified `name` field in `PathogenFactory` to use a sequence with a prefix.
- Modified `name` field in `SignalCategoryFactory` to use a sequence with a prefix.
- Modified `name` field in `SignalTypeFactory` to use a sequence with a prefix.
- Modified `name` field in `SignalFactory` to use a sequence with a prefix.
- Modified `short_description` and `description` fields in `SignalFactory` to generate fake text with a maximum of 500 characters.
- Added test cases for filtering and searching in `SignalListViewTest`.
- Added test cases for `SignalDetailViewTest`.
- Removed unnecessary code in `SignalsListView` view.

### CODE REVIEW

**Review by krivard — COMMENTED** [review]
It looks like the factories for creating fake data and the tests themselves are all mixed in with production files. Historically, we've tried to avoid putting test files in directories that get deployed to production wherever possible. Does Django permit isolating tests from core app functionality or do they have to be in the same directories as the production code?

File: src/base/factories.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from factory import Sequence
+from factory.django import DjangoModelFactory
+from faker import Faker
+
+from base.models import Link, LinkTypeChoices
+
+fake = Faker()
```
[the remark this answers is not in the document]
Comment (BrainIsDead) [reply src/base/factories.py:7]: Creates fake data for tests similar to real

**Comment by BrainIsDead** [comment]
> It looks like the factories for creating fake data and the tests themselves are all mixed in with production files. Historically, we've tried to avoid putting test files in directories that get deployed to production wherever possible. Does Django permit isolating tests from core app functionality or do they have to be in the same directories as the production code?

I can move it to tests

File: src/datasources/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class SourceSubdivisionFactory(DjangoModelFactory):
+    """
+    A factory for the SourceSubdivision model.
+    """
+    class Meta:
+        model = SourceSubdivision
+
+    name = Sequence(lambda n: f'subdicision_name_{n}')
+    display_name = Sequence(lambda n: f'subdicision_display_name_{n}')
```
Comment (krivard) [inline src/datasources/tests/factories.py:29]:
```suggestion
    name = Sequence(lambda n: f'subdivision_name_{n}')
    display_name = Sequence(lambda n: f'subdivision_display_name_{n}')
```

**Review by krivard — APPROVED** [review]
👍 structure looks good! I'm not sure how view tests work but hopefully @nolangormley can cover those (`src/signals/tests/tests.py`)

**Review by nolangormley — CHANGES_REQUESTED** [review]
Could you get tests running in CI? I can't seem to get this running on my local with docker nor through pipenv.

File: src/datasources/tests/factories.py (hunk 2 above)
Comment (BrainIsDead) [reply src/datasources/tests/factories.py]: Done

**Comment by BrainIsDead** [comment]
> Could you get tests running in CI? I can't seem to get this running on my local with docker nor through pipenv.
added `python /usr/src/signal_documentation/src/manage.py tests` to docker-compose

**Review by nolangormley — APPROVED** [review]
LGTM

**Pull request merged**